File: Pref360Control/MyDeadZoneViewer.m
//
//  MyDeadZoneViewer.m
//  360 Driver
//
//  Created by Pierre TACCHI on 21/01/15.
//

#import "MyDeadZoneViewer.h"
#import "Pref360StyleKit.h"

@implementation MyDeadZoneViewer
@synthesize val = _val;
@synthesize linked = _linked;

-(void)setVal:(double)value {
    _val = value;
    [self setNeedsDisplay:YES];
}

- (void)setLinked:(BOOL)linked {
    _linked = linked;
    [self setNeedsDisplay:YES];
}

- (void)drawRect:(NSRect)dirtyRect {
    [super drawRect:dirtyRect];
    
    [Pref360StyleKit drawDeadZoneViewerWithValue:_val / 32768 linked:_linked];
}

@end

File: Pref360Control/MyDeadZoneViewer.swift
//
//  MyDeadZoneViewer.swift
//  360 Driver
//
//  Created by C.W. Betts on 5/20/16.
//  Copyright © 2016 GitHub. All rights reserved.
//

import Cocoa

class MyDeadZoneViewer: NSView {
    var val: Double = 0.0 {
        didSet {
            needsDisplay = true
        }
    }
    var linked: Bool = false {
        didSet {
            needsDisplay = true
        }
    }

    override func drawRect(dirtyRect: NSRect) {
        super.drawRect(dirtyRect)

        Pref360StyleKit.drawDeadZoneViewerWithValue(CGFloat(val / 32768.0), linked: linked)
    }
    
}
